tsl/test/sql/continuous_aggs_invalidation.sql

```diff
@@ -619,3 +619,49 @@ INSERT INTO conditions VALUES(6, 1, 1.0);
 CALL refresh_continuous_aggregate('cond_1', 10, NULL);
 SELECT * FROM cagg_invals
 WHERE cagg_id = :cond_1_id;
+
+---------------------------------------------------------------------
+-- Test that single timestamp invalidations are expanded to buckets,
+-- and adjacent buckets merged.
+---------------------------------------------------------------------
+-- First clear invalidations in a range:
+CALL refresh_continuous_aggregate('cond_10', -20, 60);
+
+-- The following three should be merged to one range 0-29
+INSERT INTO conditions VALUES (5, 1, 1.0);
+INSERT INTO conditions VALUES (15, 1, 1.0);
+INSERT INTO conditions VALUES (25, 1, 1.0);
+-- The last one should not merge with the others
+INSERT INTO conditions VALUES (40, 1, 1.0);
+
+-- Refresh to process invalidations, but outside the range of
+-- invalidations we inserted so that we don't clear them.
+CALL refresh_continuous_aggregate('cond_10', 50, 60);
+
+SELECT mat_hypertable_id AS cond_10_id
+FROM _timescaledb_catalog.continuous_agg
+WHERE user_view_name = 'cond_10' \gset
+
+SELECT * FROM cagg_invals
+WHERE cagg_id = :cond_10_id;
+
+-- should trigger two individual refreshes
+SET client_min_messages TO DEBUG1;
+CALL refresh_continuous_aggregate('cond_10', 0, 200);
+RESET client_min_messages;
+
+-- Allow at most 5 individual invalidations per refreshe
+SET timescaledb.max_invalidations_per_refresh=5;
```

Comment:
Suggest to add test, that checks this value is correct and produce an error if not

Reply:
I can do that if deemed important. But I don't think we should waste time on that for something that is just a session variable for debugging and not an real GUC. Also, it is not possible to parse session variables when set, so the failure would appear when running refresh and persist until the variable is changed.